### ISSUE TITLE
ODataUriExtensions.BuildUri ignores $it literal

### ISSUE THREAD

**Issue by voronov-maxim** [OData/odata.net#1974]
I'm trying to use $filter query inside $expand query to filter navigation property collection. ODataUriExtensions.BuildUri does not produce $it to the uri.

### Assemblies affected
Microsoft.Odata.Core 7.8.1

### Reproduce steps
```cs
var order = new EdmEntityType("OdataToEntity", "Order");
order.AddKeys(order.AddStructuralProperty("Id", EdmPrimitiveTypeKind.String, false));
order.AddStructuralProperty("Product", EdmPrimitiveTypeKind.String, false);

var item = new EdmEntityType("OdataToEntity", "OrderItems");
item.AddKeys(item.AddStructuralProperty("Id", EdmPrimitiveTypeKind.String, false));

var itemProperty = order.AddUnidirectionalNavigation(new EdmNavigationPropertyInfo
{
    ContainsTarget = false,
    Name = "Items",
    Target = item,
    TargetMultiplicity = EdmMultiplicity.Many
});

var defaultContainer = new EdmEntityContainer("OdataToEntity", "DefaultContainer");
var orders = defaultContainer.AddEntitySet("Orders", order);
var items = defaultContainer.AddEntitySet("Items", item);
orders.AddNavigationTarget(itemProperty, items);

var model = new EdmModel();
model.AddElement(defaultContainer);

var options = new Dictionary<String, String> { { "$expand", "Items($filter=$it/Product eq 'unknown')" } };
var parser = new ODataQueryOptionParser(model, order, orders, options);
var odataUri = new ODataUri()
{
    Path = new ODataPath(),
    SelectAndExpand = parser.ParseSelectAndExpand()
};
Uri uri = odataUri.BuildUri(ODataUrlKeyDelimiter.Parentheses);
String data = Uri.UnescapeDataString(uri.OriginalString);
Debug.Assert(data == "?$expand=Items($filter=Product eq 'unknown')");
```

### Expected result
Uri must be "?$expand=Items($filter=$it/Product eq 'unknown')".

### Actual result
Uri is "?$expand=Items($filter=Product eq 'unknown')".

**Comment by mikepizzo** [comment]
Probably related to #1906.

**Comment by KenitoInc** [comment]
This is not a regression, but an existing bug.
Range Variables are handled outside the AST. So if an `expand clause` has an `explicit range variable`, we don't have a way of adding it to the Uri when we call `BuildUri` extension method.
cc @xuzhg @mikepizzo

**Comment by voronov-maxim** [comment]
@KenitoInc 
This method can detect if the property is an "explicit range variable":
```cs
public static bool IsDifferentSource(FilterClause filterClause, SingleValuePropertyAccessNode property)
{
    if (filterClause.RangeVariable is ResourceRangeVariable filterSource &&
        property.Source is SingleResourceNode propertySource)
        return filterSource.NavigationSource == propertySource.NavigationSource;

    return false;
}
```

**Comment by KenitoInc** [comment]
Fixed by #1991 and #2004
